### PR TITLE
Address #194 for nginx

### PR DESCRIPTION
**Pull request by iancward**
Update nginx integration template with features expressed in current integration.

### CODE REVIEW

File: templates/default/nginx.yaml.erb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 instances:
 <% @instances.each do |i| -%>
   - nginx_status_url: <%= i['nginx_status_url'] %>
+    <%= "ssl_validation: #{i['ssl_validation']}" unless i['ssl_validation'].nil? -%>
```
Comment (olivielpeau) [inline templates/default/nginx.yaml.erb]: IMO the indentation and format of this line could break easily if we make changes to it in the future, could you rather write:
```erb
<% unless i['ssl_validation'].nil? -%>
    ssl_validation: i['ssl_validation']
<% end -%>
```

What do you think?

Comment (iancward) [inline templates/default/nginx.yaml.erb]: I waffled on whether to do it that way, but now that I think about it, what you put above will also be more consistent with the rest of the template. Let me take care of that.

Comment (iancward) [inline templates/default/nginx.yaml.erb]: Oh! I had not even thought of chefspec; let me add that as well.

Comment (iancward) [inline templates/default/nginx.yaml.erb]: @olivielpeau I've pushed a5e1f25 with the recommended changes; I also added spec tests and updated the examples in the nginx recipe to demonstrate the new features.

**Pull request merged**